src/parser.rs

```diff
@@ -112,16 +112,19 @@ where
     where
         I::Slice: InputMatch<Pat>,
     {
-        let start = self.remaining.first();
+        let start = self.remaining.start();
         if start > self.state.last_input_pos {
             self.state.last_input_pos = start;
             self.state.expected_pats.clear();
         }
         match self.state.forest.input(self.remaining).match_left(pat) {
             Some(n) => {
-                let (matching, after, _) = self.remaining.split_at(n);
+                let (matching, after) = self
+                    .remaining
+                    .split_at::<Unknown>(/*n*/ unimplemented!())
```

Comment:
I probably need to add a trusted matching API to `indexing-str` to support this.